mxcubeweb/core/components/samplechanger.py

```diff
@@ -413,181 +408,3 @@ def _gripper_changed(self):
         self.app.server.emit(
             "queue", {"Signal": "update", "message": "all"}, namespace="/hwr"
         )
-
-
-# Disabling C901 function is too complex (19)
-def queue_mount_sample(view, data_model, centring_done_cb, async_result):  # noqa: C901
-    from mxcubeweb.app import MXCUBEApplication as mxcube
-    from mxcubeweb.routes import signals
-
-    HWR.beamline.sample_view.clear_all()
-    logging.getLogger("user_level_log").info("Loading sample ...")
-    log = logging.getLogger("user_level_log")
-
-    loc = data_model.location
-    data_model.holder_length
-
-    robot_action_dict = {
-        "actionType": "LOAD",
-        "containerLocation": loc[1],
-        "dewarLocation": loc[0],
-        "sampleBarcode": data_model.code,
-        "sampleId": data_model.lims_id,
-        "sessionId": HWR.beamline.session.session_id,
-        "startTime": time.strftime("%Y-%m-%d %H:%M:%S"),
-    }
-
-    # devices that can move sample on beam
-    # (sample changer, plate holder)
-    sample_mount_device = HWR.beamline.sample_changer
-
-    mount_from_harvester = mxcube.harvester.mount_from_harvester()
-
-    if (
-        sample_mount_device.get_loaded_sample()
-        and sample_mount_device.get_loaded_sample().get_address() == data_model.loc_str
-    ):
-        return
-
-    if hasattr(sample_mount_device, "__TYPE__"):
-        if sample_mount_device.__TYPE__ in ["Marvin", "CATS"]:
-            element = "%d:%02d" % loc
-            sample = {"location": element, "sampleID": element}
-            mxcube.sample_changer.mount_sample_clean_up(sample)
-        else:
-            sample = {
-                "location": data_model.loc_str,
-                "sampleID": data_model.loc_str,
-            }
-
-            # in this case the sample changer takes the sample from an Harvester
-            # We Harvest the sample and make it ready to load
-            if mount_from_harvester:
```

Comment:
this harvester  logic need to be ported here 
https://github.com/mxcube/mxcubecore/pull/971  right ?

Reply:
Actually, it would perhaps be better if this was integrated in the harvester or the sample changer so that we don't need to add the extra logic for handling the harvester in this method.

Maybe int this case it would be better to "proxy" the sample changer via the harvester and like this the harvester is the sample changer. It would make the entire configuration easier

Reply:
Agreed, and i think it would be better to do it right away in the mxcubecore related PR.
this one :  https://github.com/mxcube/mxcubecore/pull/971

Reply:
We would need to add `queue_harvest_sample` and `queue_harvest_next_sample` to https://github.com/mxcube/mxcubecore/blob/14fa9ddfca44dbed45785651ca85e1f0ef928afd/mxcubecore/HardwareObjects/EMBLFlexHarvester.py#L141. I could do something but it would probably be better if you could help me out as you know how the harvester works a bit better.

Reply:
OK, 
good  I remove them from this pr.

Reply:
Thanks @jbflo